src/util/tokenize.jl

```diff
@@ -0,0 +1,67 @@
+# ========================
+# A Tokenizer for Strings
+# ========================
+
+# Ben J. Ward, 2016
+
+
+module Tokenize
+
+export
+    Tokenizer, TokenizerResult, tokenize
+
+
+"""
+Tokenizer type that is responsible for splitting a string into token according to a
+regex specification. The regex specification is stored in `dict`.
+
+**Fields:**
+
+* `dict`:      A Dictionary of String keys and Regex values.
+* `tokenizer`: A regex that is generated from the specification.
+"""
+type Tokenizer
+    dict::Dict{ASCIIString, Regex}
```

Comment:
Will using String work in 0.4 though? I understand that String used to be AbstractString or something like that.

```
julia> String
WARNING: Base.String is deprecated, use AbstractString instead.
  likely near no file:0
AbstractString
```